sycl/include/sycl/bit_cast.hpp

```diff
@@ -41,8 +41,8 @@ constexpr
 #if __has_builtin(__builtin_bit_cast)
   return __builtin_bit_cast(To, from);
 #else  // __has_builtin(__builtin_bit_cast)
-  static_assert(std::is_trivially_default_constructible<To>::value,
-                "To must be trivially default constructible");
+  static_assert(std::is_default_constructible<To>::value,
+                "To must be default constructible");
```

Comment:
Note for reviewers: I honestly don't know why would we need a trivial constructor here, but if necessary, we can return that check back and add a special case for `half` like we used to have, see 909a8249b73012d2da47f76b9b49ae46a7f236e6